src/toutaticeClient.js

```diff
@@ -28,6 +29,48 @@ class ToutaticeClient {
     if (response.status === 200) return response.json()
     else return null
   }
+
+  async getApps() {
+    log('info', 'getApps begins')
+    const response = await fetch(
+      'https://partenaires.ipanema.education.fr/safran/api/v1/catalogues/3ec0316e-2cbc-4a7e-ba0d-81e127d98600/sync',
+      {
+        method: 'POST',
+        headers: {
+          Authorization: `Bearer ${this.token}`,
+          'Content-Type': 'application/json'
+        },
+        body: '{}'
+      }
+    )
+    if (response.status === 202) {
+      log('info', 'Syncing apps ...')
+      await this.sleep(5000)
```

Comment:
What? 

I imagine that the `/sync` endpoint is returning a stream, right? 

If yes, you can use node-fetch correctly to handle the stream https://github.com/node-fetch/node-fetch#streams  

No more `sleep` needed or else.

Reply:
```
import fetch from 'node-fetch';

const read = async body => {
	let error;
	body.on('error', err => {
		error = err;
	});

	for await (const chunk of body) {
		console.dir(JSON.parse(chunk.toString()));
	}

	return new Promise((resolve, reject) => {
		body.on('close', () => {
			error ? reject(error) : resolve();
		});
	});
};

try {
	const response = await fetch('https://partenaires.ipanema.education.fr/safran/api/v1/catalogues/3ec0316e-2cbc-4a7e-ba0d-81e127d98600/sync');
const data = 	await read(response.body);
console.log({data})
} catch (err) {
	console.error(err.stack);
}
```

Can you try that?

Reply:
It didn't work, the konnector is stoping after returning the 'new Promise', never getting into the 'for await' loop and 
the next API call is not made too, as it act like the konnector has been fully run without any issue

Reply:
I send you in private de doc I work from, trello's card apart